src/parser.h

```diff
@@ -9,21 +9,21 @@ namespace ada {
 
   public:
     Parser(std::string_view input, std::optional<ada::URL> optional_base_url, std::optional<ada::encoding_type> encoding, std::optional<ada::state> state);
-    ada::URL getURL();
+    ada::URL get_url();
 
   private:
-    void parseState();
+    void parse_state();
 
-    std::string_view input;
-    std::string buffer;
-    bool at_sign_seen = false;
-    bool inside_brackets = false;
-    bool password_token_seen = false;
+    std::string_view input{};
+    std::string buffer{};
+    bool at_sign_seen{};
+    bool inside_brackets{};
+    bool password_token_seen{};
 
     const char* pointer;
 
-    ada::encoding_type encoding = ada::encoding_type::UTF8;
-    ada::state state = SCHEME_START;
+    ada::encoding_type encoding{ada::encoding_type::UTF8};
+    ada::state state{SCHEME_START};
     std::optional<ada::state> state_override;
 
     ada::URL url;
```

Comment:
If you look at the C++ standard library, it uses lower case even for names that are normally capitalized (e.g., UTF). https://en.cppreference.com/w/cpp/locale/codecvt_utf8 So one should reflect as to whether a class 'URL' shouldn't be named 'url'. (This is not a request for change!!!)

Reply:
My thought with `class`es was to have capital letters. I think this can be `ada::Url`

Reply:
> My thought with classes was to have capital letters. I think this can be ada::Url

A sensible reference might be the url class in the Boost library. I do not use Boost myself, but it is old, well established and quite active:

https://www.boost.org/doc/libs/1_81_0/libs/url/doc/html/url/ref/boost__urls__url.html

Please see the classes used in the standard library...

https://en.cppreference.com/w/cpp/filesystem



path(C++17) | represents a path  (class)
-- | --
filesystem_error(C++17) | an exception thrown on file system errors (class)
directory_entry(C++17) | a directory entry (class)
directory_iterator(C++17) | an iterator to the contents of the directory (class)
recursive_directory_iterator(C++17) | an iterator to the contents of a directory and its subdirectories (class)
file_status(C++17) | represents file type and permissions (class)
space_info(C++17) | information about free and available space on the filesystem (class)
file_type(C++17) | the type of a file (enum)
perms(C++17) | identifies file system permissions (enum)
perm_options(C++17) | specifies semantics of permissions operations (enum)
copy_options(C++17) | specifies semantics of copy operations (enum)
directory_options(C++17) | options for iterating directory contents (enum)


Notice how they are named?

Reply:
Let's follow it. I'm convinced. Just committed with the changes.